### PR TITLE
drivers: sensor: Driver for HTU21D temp/humidity sensor

### PR DESCRIPTION
**Pull request by JonasNorling**
Driver for TE Connectivity MEAS HTU21D temperature and humidity sensor.

The driver was tested on a custom ESP32 board.

### CODE REVIEW

**Review by sylvioalves — APPROVED** [review]
@JonasNorling got myself in here. LGTM but I have not tested it.  
Can you add yourself as owner of this source in CODEOWNERS file?

File: dts/bindings/sensor/meas,htu21d.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2021, Jonas Norling
+# SPDX-License-Identifier: Apache-2.0
+
+description: TE Connectivity HTU21D temperature and humidity sensor
+
+compatible: "meas,htu21d"
```
Comment (mbolivar-nordic) [inline dts/bindings/sensor/meas,htu21d.yaml:6]: `meas` is the vendor prefix for Measurement Specialties, but the vendor in the description is "TE Connectivity".

This looks like a violation of the rules: https://docs.zephyrproject.org/latest/guides/dts/bindings.html#vendor-prefixes

What's going on?

File: drivers/sensor/htu21d/htu21d.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+struct htu21d_data {
+	const struct device *i2c;
+	uint16_t t_sample;
+	uint16_t rh_sample;
+};
+
+struct htu21d_config {
+	const char *i2c_bus;
+	uint16_t i2c_addr;
+};
```
Comment (mbolivar-nordic) [inline drivers/sensor/htu21d/htu21d.h:25]: Just use an i2c_dt_spec, please.

File: drivers/sensor/htu21d/htu21d.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+static struct htu21d_data htu21d_driver;
+static const struct htu21d_config htu21d_config = {
+	.i2c_bus = DT_INST_BUS_LABEL(0),
+	.i2c_addr = DT_INST_REG_ADDR(0),
+};
+
+DEVICE_DT_INST_DEFINE(0, htu21d_init, NULL,
+		      &htu21d_driver, &htu21d_config, POST_KERNEL,
+		      CONFIG_SENSOR_INIT_PRIORITY, &htu21d_driver_api);
```
Comment (mbolivar-nordic) [inline drivers/sensor/htu21d/htu21d.c:177]: Sensors which are I2C/SPI bus devices must have multi-instance drivers (per @MaureenHelm; sensor driver maintainer); a single-instance driver like this is not allowed in new code.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Thanks for your patch. A few DT/driver model issues from my side.

File: dts/bindings/sensor/meas,htu21d.yaml (hunk 1 above)
Comment (JonasNorling) [reply dts/bindings/sensor/meas,htu21d.yaml:6]: Measurement Specialties was bought by TE Connectivity last decade. They seem to use both names, in addition to "MEAS", when marketing this range of sensors. meas in in the vendor file, both in Zephyr and Linux, whereas TE is not, so it felt appropriate to keep using that prefix.
Updated yaml file to include both names.

File: drivers/sensor/htu21d/htu21d.h (hunk 2 above)
Comment (JonasNorling) [reply drivers/sensor/htu21d/htu21d.h:25]: done

File: drivers/sensor/htu21d/htu21d.c (hunk 3 above)
Comment (JonasNorling) [reply drivers/sensor/htu21d/htu21d.c:177]: done

**Comment by mbolivar-nordic** [comment]
@JonasNorling I'm sorry this was closed before I had a chance to revisit. The merge window for v3.1 is reopening soon, so please rebase and reopen this pull request if you'd like to try again.

**Pull request closed without merging**